### ISSUE TITLE
Predicter seems not working most of the time

### ISSUE THREAD

**Issue by rogercooper76** [jakeret/tf_unet#150]
I copied the codes from demo_toy_problem.ipynb, and ran it with my machine. Based on the result, I felt that the predicter (or model restore) function does not work in most of time.

The 'Verification error' value did not change through the training, and the prediction result of final test figure is complete black. The Tensorflow version I am using is 1.4.0. Training log, tensorboard figure, last prediction figure, and final test figure are attached below. Are those expected? If not, does anyone know what I might do wrong? Thanks in advance.

[demo_toy_problem_log.txt](https://github.com/jakeret/tf_unet/files/1658380/demo_toy_problem_log.txt)

![demo_toy_porblem_tensorboard](https://user-images.githubusercontent.com/19596166/35312993-f99a34da-0072-11e8-8c8a-85e4479d005e.png)

![epoch_9](https://user-images.githubusercontent.com/19596166/35313131-90f214c4-0073-11e8-8a6a-973375f50ecf.jpg)

![toy_problem](https://user-images.githubusercontent.com/19596166/35312978-e789371e-0072-11e8-97ef-6ed83df7de7b.png)

**Comment by jakeret** [comment]
Have you checked the values of the prediction. Wondering if this might be a plotting issue.
Based on the other figures, the training looks quite good

**Comment by rogercooper76** [comment]
Thanks for the reply, @jakeret. Based on the log data that I attached, the verification also got stuck to the same value through the training. Since I am still a newbie of TF, please let me know if there is other way to check the prediction value. Thanks. (PS. I am using Python 3.5 if it matters)

**Comment by jakeret** [comment]
Is this reproducible i.e. does it always get stuck?

**Comment by rogercooper76** [comment]
Sorry for the late reply. Yes, it is reproducible. Roughly speaking, only one out of tens runs generates meaningful prediction outcome in the final test.

**Comment by jakeret** [comment]
I played around a little bit with the demo but it seemed to work fine on my machine. Not sure whats happening in your case...

**Comment by rogercooper76** [comment]
I get it. Still, thanks for your time and help, @jakeret .

**Comment by shilpa-ananth** [comment]
I had a similar issue when trying to predict on my own data. Try predicting by loading the data through the data provider itself, and then feeding it to the function rather than directly feeding your data to the unet.predict() function and see if it helps.

Another issue that I faced was that matplotlib normalizes different images differently. So when viewing multiple results they were all normalized differently. I just concatenated all the results together so that I could view them on a relative scale.
